### PR TITLE
Create default EE in seperate awx-managment cmd

### PR DESCRIPTION
**Pull request by kdelee**
Create EE at a seperate time and also attach a registry credential if
auth information provided

### CODE REVIEW

**Comment by jbradberry** [comment]
@kdelee do we have any associated PRs that make use of this new command during install or upgrade?

**Review by jbradberry — APPROVED** [review]

**Comment by jbradberry** [comment]
This PR addresses the problem in #10287.  Also, #9851, but we need to change the packaging to invoke this command.

**Comment by kdelee** [comment]
@jbradberry I've got  a PR for the downstream traditional installer, but we will need to make a PR to the operator to use this, except in the operator we may need to some extra finess around the secrets because we won't be able to directly pass secrets to this command

for operator:
 - need to call the register_default_execution_environment commands
 - for ee that need auth, we need to pull secret from an openshift secret, see use of k8s_info for other secrets like the SECRET_KEY
 - https://github.com/ansible/awx-operator/blob/devel/roles/installer/tasks/broadcast_websocket_configuration.yml#L10-L15

**Comment by Zokormazo** [comment]
Since `DEFAULT_EXECUTION_ENVIRONMENTS` is a list, I went with 2 ees there and ended with two credentials named as `Default Execution Environment Registry Credential` . Since we are not allowing to set a credential per image, could we just reuse the first generated one?

**Comment by kdelee** [comment]
@Zokormazo thats strange, did you call the command twice? I was hoping using `update_or_create` on the credential would prevent that. Let me try on my side again

**Comment by kdelee** [comment]
@Zokormazo you were right :D I fixed my use of `update_or_create` and it should work now. Can you try it out?

File: awx/main/management/commands/register_default_execution_environments.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+
+class Command(BaseCommand):
+    """Create default execution environments, intended for new installs"""
+
+    help = """
+    Creates or updates the execution environments set in settings.DEFAULT_EXECUTION_ENVIRONMENTS if they are not yet created.
+    Optionally provide authentication details to create or update a container registry credential that will be set on all of these default execution environments.
+    Note that settings.DEFAULT_EXECUTION_ENVIRONMENTS is and ordered list, the first in the list will be used for project updates and system jobs.
+    """
```
Comment (kdelee) [inline awx/main/management/commands/register_default_execution_environments.py:19]: @shanemcd @jbradberry I updated this help text, what do you think?

**Review by shanemcd — APPROVED** [review]

File: awx/api/serializers.py [hunk 2]
```diff
@@ -2603,8 +2603,6 @@ def get_summary_fields(self, obj):
         return summary_dict
 
     def validate(self, attrs):
-        if self.instance and self.instance.managed_by_tower:
-            raise PermissionDenied(detail=_("Modifications not allowed for managed credentials"))
```
Comment (kdelee) [inline awx/api/serializers.py]: @jbradberry mentioned I should revert this

File: awx/main/management/commands/register_default_execution_environments.py (hunk 1 above)
Comment (jbradberry) [reply awx/main/management/commands/register_default_execution_environments.py:19]: "... is an ordered list"

**Review by rebeccahhh — APPROVED** [review]
I like this a lot

File: awx/main/management/commands/register_default_execution_environments.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        )
+        parser.add_argument(
+            "--registry-password",
+            type=str,
+            default="",
+            help="Password or token for CLI authentication with the container registry",
+        )
+        parser.add_argument(
+            "--verify-ssl",
+            type=lambda x: bool(strtobool(str(x))),
```
Comment (kdelee) [inline awx/main/management/commands/register_default_execution_environments.py:48]: nice

**Pull request merged**